alvr/xtask/src/dependencies.rs

```diff
@@ -94,10 +94,11 @@ pub fn build_ffmpeg_linux() -> std::path::PathBuf {
             "--disable-network",
             "--enable-lto",
             format!(
-                "--disable-everything {} {} {} {}",
+                "--disable-everything {} {} {} {} {}",
+                "--enable-encoder=h264_nvenc --enable-encoder=hevc_nvenc --enable-nonfree --enable-nvenc --enable-cuda --enable-cuda-nvcc --enable-libnpp --nvccflags=\"-gencode arch=compute_52,code=sm_52 -O2\" --extra-cflags=-I/usr/local/cuda/include/  --extra-ldflags=-L/usr/local/cuda/lib64/",
```

Comment:
This line should use pkg-config to get cuda path and not hardcode /usr/local.

Also can you explain --nvccflags=\"-gencode arch=compute_52,code=sm_52 -O2\"

Reply:
With first to me need a bit help.

The second is nvcc by default use com_30 and sm_30 which is too old cards to use nvenc 
So nvenc is available from Maxwell 
https://developer.nvidia.com/nvidia-video-codec-sdk (Supported Format Details)
And matrix is shown minimal is 52 to use nvenc https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/  Also the param is written in doc of Nvidia how to build nvenc (sorry I'm loss the link to pdf)

Reply:
On pkg-config, I expect ffmpeg to already include it in its build system, in which case you just need to set the `PKG_CONFIG_PATH` variable in case cuda is not installed in the standard location (i.e. /usr/lib/pkgconfig/cuda.pc is not present).
In this case, nothing should be done in the build script itself and adding that as an instruction in the wiki seems enough.

If having the correct `PKG_CONFIG_PATH` doesn't work, you should use it manually with https://docs.rs/pkg-config/0.3.9/pkg_config/ where you can get the value. I think you should set https://docs.rs/pkg-config/0.3.9/pkg_config/struct.Config.html#method.cargo_metadata to false, because you just want to access the values and not link with cuda from the rust file.

And for the nvcc_flags, some comment like that in the code would be enough to understand why this magic value

Reply:
https://docs.nvidia.com/video-technologies/video-codec-sdk/ffmpeg-with-nvidia-gpu/index.html#commonly-faced-issues-and-tips-to-resolve-them